bucket.go

```diff
@@ -0,0 +1,84 @@
+package turtleDB
+
+import "sync"
+
+func newBucket() *bucket {
+	var b bucket
+	b.m = make(map[string]Value)
+	return &b
+}
+
+// bucket is thread-safe a basic data store
+type bucket struct {
+	mux sync.RWMutex
+
+	m map[string]Value
+}
+
+// get will retrieve a value for a provided key
+func (b *bucket) get(key string) (value Value, err error) {
+	var ok bool
+	b.mux.RLock()
+	if value, ok = b.m[key]; !ok {
+		// Value does not exist for this key
+		err = ErrKeyDoesNotExist
+	}
+	b.mux.RUnlock()
+	return
+}
+
+// put will set a value for a provided key
+func (b *bucket) put(key string, value Value) {
+	b.mux.Lock()
+	b.m[key] = value
+	b.mux.Unlock()
+}
+
+// delete will remove a provided key
+func (b *bucket) delete(key string) (ok bool) {
+	b.mux.Lock()
+	if _, ok = b.m[key]; ok {
```

Comment:
Technically, checking if it exists first is a bit faster. I believe it's faster up to a 50% miss ratio. I guess at the end of the day it won't really affect performance and it will clean up the code a bit. I'll go ahead and simplify it.